### PR TITLE
feat(core): finalize migration-03 auth removal and 3-core slim package (#317)

### PR DESCRIPTION
**Pull request by djm81**
# Description

Complete `module-migration-03-core-slimming` by executing task 10.6 after `backlog-auth-01` landed.

This PR removes the core `auth` module from `specfact-cli`, keeps central token utilities in core, and routes auth UX to backlog bundle entrypoints (`specfact backlog auth ...`). It also updates OpenSpec artifacts/specs/tasks and docs to align with the final 3-core model.

**Fixes** #317

**New Features** #340

**Contract References**:
- Existing public APIs remain contract-protected; no new public contract surface added in this increment.
- Updated behavior validated against OpenSpec deltas in `core-lean-package` and `profile-presets`.

## Type of Change

Please check all that apply:

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [x] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [ ] 🔒 Contract enforcement (adding/updating `@icontract` decorators)
- [x] 🧪 Test enhancement (scenario tests, property-based tests)
- [ ] 🔧 Refactoring (code improvement without functionality change)

## Contract-First Testing Evidence

**Required for all changes affecting CLI commands or public APIs:**

### Contract Validation

- [ ] **Runtime contracts added/updated** (`@icontract` decorators on public APIs)
- [ ] **Type checking enforced** (`@beartype` decorators applied)
- [ ] **CrossHair exploration** completed: `hatch run contract-test-exploration`
- [ ] **Contract violations** reviewed and addressed

### Test Execution

- [ ] **Contract validation**: `hatch run contract-test-contracts` ✅
- [ ] **Contract exploration**: `hatch run contract-test-exploration` ✅
- [ ] **Scenario tests**: `hatch run contract-test-scenarios` ✅
- [ ] **Full test suite**: `hatch run contract-test-full` ✅

### Test Quality

- [x] **CLI commands tested** with typer test client
- [ ] **Edge cases covered** with Hypothesis property tests
- [x] **Error handling tested** with invalid inputs
- [x] **Rich console output verified** manually or with snapshots

## How Has This Been Tested?

**Contract-First Approach**: OpenSpec TDD order was followed for 10.6 with failing-before and passing-after evidence captured in `openspec/changes/module-migration-03-core-slimming/TDD_EVIDENCE.md`.

### Manual Testing

- [x] Tested CLI commands manually
- [x] Verified rich console output
- [x] Tested with different input scenarios
- [x] Checked error messages for clarity

### Automated Testing

- [x] Contract validation passes
- [ ] Property-based tests cover edge cases
- [x] Scenario tests cover user workflows
- [x] All existing tests still pass

### Test Environment

- Python version: 3.11
- OS: Linux (worktree CI-like environment)

## Checklist

- [x] My code follows the style guidelines (PEP 8, ruff format, isort)
- [x] I have performed a self-review of my code
- [ ] I have added/updated contracts (`@icontract`, `@beartype`)
- [x] I have added/updated docstrings (Google style)
- [x] I have made corresponding changes to documentation
- [x] My changes generate no new warnings (basedpyright, ruff, pylint)
- [x] All tests pass locally
- [x] I have added tests that prove my fix/feature works
- [x] Any dependent changes have been merged

## Quality Gates Status

- [x] **Type checking** ✅ (`hatch run type-check`)
- [ ] **Linting** ✅ (`hatch run lint`)
- [ ] **Contract validation** ✅ (`hatch run contract-test-contracts`)
- [ ] **Contract exploration** ✅ (`hatch run contract-test-exploration`)
- [ ] **Scenario tests** ✅ (`hatch run contract-test-scenarios`)

## Screenshots/Recordings (if applicable)

N/A (CLI/output behavior change only).

### CODE REVIEW

File: src/specfact_cli/commands/auth.py [hunk 1]
```diff
@@ -1,6 +0,0 @@
-"""Backward-compatible app shim. Implementation moved to modules/auth/."""
-
-from specfact_cli.modules.auth.src.commands import app
-
```
[the remark this answers is not in the document]
Comment (djm81) [reply src/specfact_cli/commands/auth.py]: The auth.py import will be removed from all core cli modules for consistency.
